### PR TITLE
STORM-2833: Use the original host name when closing (2.x)

### PR DESCRIPTION
**Pull request by revans2**

### CODE REVIEW

File: storm-client/src/jvm/org/apache/storm/messaging/netty/Client.java [hunk 1]
```diff
@@ -82,6 +82,10 @@
     private final ClientBootstrap bootstrap;
     private final InetSocketAddress dstAddress;
     protected final String dstAddressPrefixedName;
+    //The actual name of the host we are trying to connect to so that
+    // when we remove ourselves from the connection cache there is no concern that
+    // the resolved host name is different.
+    private final String host;
```
Comment (srdo) [inline storm-client/src/jvm/org/apache/storm/messaging/netty/Client.java]: Nit: Rename to dstHost or something like that to make it clear which host it is without having to read the comment.

**Comment by srdo** [comment]
+1

**Comment by revans2** [comment]
@srdo I made the name change you suggested.

**Comment by srdo** [comment]
Thanks, still +1

**Comment by HeartSaVioR** [comment]
Addressed by 107a9c9

**Pull request closed without merging**